cmd/osbuild-deploy-container/main.go

```diff
@@ -0,0 +1,181 @@
+	var repoData map[string][]rpmmd.RepoConfig
+	err := json.Unmarshal([]byte(reposStr), &repoData)
+	if err != nil {
+		fail(fmt.Sprintf("error loading repositories: %s", err))
+	}
+	archRepos, ok := repoData[arch]
+	if !ok {
+		fail(fmt.Sprintf("no repositories defined for %s", arch))
+	}
+	return archRepos
+}
+
+func loadConfig(path string) BuildConfig {
+	fp, err := os.Open(path)
+	check(err)
+	defer fp.Close()
+
+	dec := json.NewDecoder(fp)
+	dec.DisallowUnknownFields()
+	var conf BuildConfig
+
+	check(dec.Decode(&conf))
+	if dec.More() {
+		fail(fmt.Sprintf("multiple configuration objects or extra data found in %q", path))
+	}
+	return conf
+}
+
+func makeManifest(imageref string, config *BuildConfig, repos []rpmmd.RepoConfig, archName string, seedArg int64, cacheRoot string) (manifest.OSBuildManifest, error) {
+	manifest, err := Manifest(imageref, config, repos, archName, seedArg)
+	check(err)
+
+	// depsolve packages
+	solver := dnfjson.NewSolver(modulePlatformID, releaseVersion, archName, distroName, cacheRoot)
+	solver.SetDNFJSONPath("./dnf-json")
+	depsolvedSets := make(map[string][]rpmmd.PackageSpec)
+	for name, pkgSet := range manifest.GetPackageSetChains() {
+		res, err := solver.Depsolve(pkgSet)
+		if err != nil {
+			return nil, err
+		}
+		depsolvedSets[name] = res
+	}
+
+	// resolve container
+	resolver := container.NewResolver(archName)
+	containerSpecs := make(map[string][]container.Spec)
+	for plName, sourceSpecs := range manifest.GetContainerSourceSpecs() {
+		for _, c := range sourceSpecs {
+			resolver.Add(c)
+		}
+		containerSpecs[plName], err = resolver.Finish()
+		if err != nil {
+			return nil, err
+		}
+	}
+	mf, err := manifest.Serialize(depsolvedSets, containerSpecs, nil)
+	if err != nil {
+		fail(fmt.Sprintf("[ERROR] manifest serialization failed: %s", err.Error()))
+	}
+	return mf, nil
+}
+
+func saveManifest(ms manifest.OSBuildManifest, fpath string) error {
+	b, err := json.MarshalIndent(ms, "", "  ")
+	if err != nil {
+		return fmt.Errorf("failed to marshal data for %q: %s\n", fpath, err.Error())
+	}
+	b = append(b, '\n') // add new line at end of file
+	fp, err := os.Create(fpath)
+	if err != nil {
+		return fmt.Errorf("failed to create output file %q: %s\n", fpath, err.Error())
+	}
+	defer fp.Close()
+	if _, err := fp.Write(b); err != nil {
+		return fmt.Errorf("failed to write output file %q: %s\n", fpath, err.Error())
+	}
+	return nil
+}
+
+func main() {
+
+	arch := common.CurrentArch()
+	repos := loadRepos(arch)
+
+	var outputDir, osbuildStore, rpmCacheRoot, configFile, imageref string
+	flag.StringVar(&outputDir, "output", ".", "artifact output directory")
+	flag.StringVar(&osbuildStore, "store", ".osbuild", "osbuild store for intermediate pipeline trees")
+	flag.StringVar(&rpmCacheRoot, "rpmmd", "/var/cache/osbuild/rpmmd", "rpm metadata cache directory")
+	flag.StringVar(&configFile, "config", "", "build config file")
+	flag.StringVar(&imageref, "imageref", "", "container image to deploy")
```

Comment:
Might be better to rename this to `--imgref` to match the argument in `rpm-ostree container image deploy ...`.